lib/archethic/oracle_chain.ex

```diff
@@ -186,4 +190,82 @@ defmodule Archethic.OracleChain do
       |> DateTime.from_naive!("Etc/UTC")
     end
   end
+
+  @doc """
+    Methods Serves to resolve Situation
+    time  : event
+    12:00 : auth node rejoins
+    12:01:40 : completes sync ,recieves oracle summary to be verified by this auth node
+    12:01:55 : will result failure as last oracle_gen_addr not there
+  """
+  @spec update_last_summ_gen_addr :: :ok
+  def update_last_summ_gen_addr() do
+    summary_interval =
+      Application.get_env(:archethic, Archethic.OracleChain.Scheduler)
+      |> Keyword.fetch(:summary_interval)
+      |> elem(1)
+
+    current_time = DateTime.truncate(DateTime.utc_now(), :second)
+
+    cron_expression = Crontab.CronExpression.Parser.parse!(summary_interval, true)
+
+    naive_date =
+      current_time
+      |> DateTime.truncate(:second)
+      |> DateTime.to_naive()
+
+    last_summary_date =
+      if Crontab.DateChecker.matches_date?(cron_expression, naive_date) do
+        DateTime.truncate(current_time, :second)
+      else
+        cron_expression
+        |> Crontab.Scheduler.get_previous_run_date!(naive_date)
+        |> DateTime.from_naive!("Etc/UTC")
+      end
+
+    MemTable.put_addr(Crypto.derive_oracle_address(last_summary_date, 0), last_summary_date)
+    :ok
+  end
+
+  @doc """
+  Updates ets table with current_summary_gen_addr
+  Address: genesis address of current(summarydate) oracle chain
+  DateTime: current summary datetime
+  """
+  @spec update_curr_summ_gen_addr :: :ok
+  def update_curr_summ_gen_addr() do
```

Comment:
Instead of calculating the next summary_date, you can just use `OracleChain.next_summary_date()`